sklearn/isotonic.py

```diff
@@ -271,6 +272,7 @@ def _build_y(self, X, y, sample_weight, trim_duplicates=True):
         check_consistent_length(X, y, sample_weight)
         X, y = [check_array(x, ensure_2d=False) for x in [X, y]]
 
+        X = as_float_array(X)
```

Comment:
in logistic regression it seems like the check_array way is used:

https://github.com/Henley13/scikit-learn/blob/0a5eda04b90c5b89c14f870bfde45ed71f89311c/sklearn/linear_model/sag.py#L239